test/unit/math/rev/mat/prob/bernoulli_logit_glm_lpmf_test.cpp

```diff
@@ -380,3 +380,31 @@ TEST(ProbDistributionsPoissonLogGLM,
   EXPECT_THROW(stan::math::bernoulli_logit_glm_lpmf(y, x, alpha, betaw2),
                std::domain_error);
 }
+
+TEST(ProbDistributionsBernoulliLogitGLM, test_scalar_rowvec_stdvec) {
+  Matrix<int, Dynamic, 1> y(3, 1);
```

Comment:
Stan compiler can not generate types `Matrix<int, Dynamic, 1>` or `Matrix<int, 1, Dynamic>`. When `vector` or `rowvector` stan types are compiled to C++, they are always mapped to `Matrix<>`  that contain `double`s. Only array in stan can contain integers. That is `std::vector` in C++. (That is why I opened #1261)

So there is probably no need to add these tests for y that is an integer. While those are no harm, they do little good for for use of Stan Math in Stan. Instead such tests should be added for alpha in bernoulli_logit_glm and poisson_log_glm and  alpha and phi in neg_binomial_2_log_glm. Only for normal_id_glm y is double so these tests are actually needed. But in normal_id_glm such tests are also needed for alpha and sigma.

Reply:
thanks, I'll remove those tests. Should I be testing Eigen::Array for the response as well?